### PR TITLE
refactor(BA-3516): Integrate `ModifyScalingGroup` action with existing GQL resolver

### LINKED ISSUE #7523: Integrate ModifyScalingGroup Action with existing GraphQL implementation
## Objective

  Complete Modify action with GraphQL integration and comprehensive tests.

## Implementation Details

  **GraphQL mutation modification (~30 lines)**

- `ModifyScalingGroupInput` → `Updater` conversion logic
- `simple_db_mutate_returning_item` → `processors.scaling_group.modify.wait_for_complete()`

  **ScalingGroupOpts conversion logic cleanup (~20 lines)**

- `OptionalState` / `TriState` conversion for nullable/non-nullable fields

  **Additional Unit tests:**

- `test_modify_scaling_group_service.py` (~80 lines)
- `test_modify_scaling_group_repository.py` (~100 lines)
- Edge case tests (not found, validation, etc.) (~50 lines)

## Estimated Lines

  ~280 lines

## Dependencies

- Story: ModifyScalingGroup Action - Updater Spec

## Acceptance Criteria

- [ ] GraphQL mutation uses new processor pattern
- [ ] Input conversion properly handles all fields including ScalingGroupOpts
- [ ] `OptionalState` / `TriState` correctly applied to nullable vs non-nullable fields
- [ ] Edge cases tested (scaling group not found, invalid inputs)
- [ ] All unit tests pass
- [ ] Legacy `simple_db_mutate_returning_item` removed from `ModifyScalingGroup`

JIRA Issue: BA-3516

### PR DESCRIPTION
**Pull request by jopemachine**
resolves #7523 (BA-3516)

**Checklist:** (if applicable)

- [ ] Milestone metadata specifying the target backport version
- [x] Mention to the original issue
- [ ] Installer updates including:
  - Fixtures for db schema changes
  - New mandatory config options
- [ ] Update of end-to-end CLI integration tests in `ai.backend.test`
- [ ] API server-client counterparts (e.g., manager API -> client SDK)
- [x] Test case(s) to:
  - Demonstrate the difference of before/after
  - Demonstrate the flow of abstract/conceptual models with a concrete implementation
- [ ] Documentation
  - Contents in the `docs` directory
  - docstrings in public interfaces and type annotations

### CODE REVIEW

File: src/ai/backend/manager/api/gql_legacy/scaling_group.py [hunk 1]
```diff
@@ -608,6 +620,77 @@ class ModifyScalingGroupInput(graphene.InputObjectType):
...
+                OptionalState.update(self.use_host_network)
+                if self.use_host_network is not None
+                else OptionalState.nop()
+            ),
+        )
+        driver_spec = ScalingGroupDriverConfigUpdaterSpec(
+            driver=(
+                OptionalState.update(self.driver)
+                if self.driver is not None
+                else OptionalState.nop()
```
Comment (HyeockJinKim) [inline src/ai/backend/manager/api/gql_legacy/scaling_group.py]: Could you take a look? We were using Undefined from GraphQL in that nop operation. @jopemachine

**Review by HyeockJinKim — CHANGES_REQUESTED** [review]

**Review by seedspirit — APPROVED** [review]
